shared/diveIntoCode/rust.md

```diff
@@ -0,0 +1,61 @@
+#### Let's dive into the code
+<!-- project link: https://github.com/resin-io-projects/resin-rust-hello-world -->
+So in the root directory of our project we see a number of files, the most important ones to focus on are:-
+* `Dockerfile.template` : This is basically a recipe file on how to build and run our application container.
+* `src/main.rs` : This is the entry point to our application code and is where all the fun happens!
+
+The most important part of a resin.io project repo is usually the `Dockerfile` or `Dockerfile.template`. The `.template` version allows you to define template variables like `%%RESIN_MACHINE_NAME%%` which enables you to push the same repository to multiple different architecture fleets.
+
+If we look at our `Dockerfile.template`, the first thing we see is:
+```
+FROM resin/%%RESIN_MACHINE_NAME%%-debian
```

Comment:
we need to add a warning that this will break for rpi1 devices because there is no `-debian` base for raspberry pi 1 & ZERO devices